### ISSUE TITLE
All main `<section>` tags are erroneously hidden with the new mobile html endpoint, and so inaccessible to readers that cannot run JS

### ISSUE THREAD

**Issue by Jaifroid** [openzim/mwoffliner#2074]
(Originally reported in https://github.com/openzim/zim-requests/issues/1129#issuecomment-2267448220)

Possibly as a remnant of the removal of details-summary tags for openeing and closing articles (#1915), all main `<section>` tags have the attribute `style="display: none;"` (see screenshot at bottom) in ZIMs scraped with the new mobile html endpoint. There is probably a piece of JS included with the ZIM (a script) that is in charge of removing this attribute (it can be seen when loading a page: the lede loads first, and then after a pause the rest of the article is unhidden). However, this logic is the wrong way round: all sections should be visible by default, and then if the reader wishes to hide a section, it can run said script.

This has a serious consequence in readers that block JS from the ZIM, or which cannot in fact run JS from the ZIM: most content in an article becomes inaccessible and cannot be unhidden. This is the case of Restricted Mode (aka JQuery Mode) in Kiwix JS and Kiwix PWA.

For relevant older issues, see https://github.com/openzim/mwoffliner/issues/962, https://github.com/openzim/mwoffliner/issues/838, https://github.com/openzim/mwoffliner/issues/952, (related) https://github.com/openzim/mwoffliner/issues/1033, and https://github.com/openzim/mwoffliner/issues/1915.

![image](https://github.com/user-attachments/assets/c175905d-93ea-4d90-9277-6e1d99d27152)

**Comment by audiodude** [comment]
@Jaifroid So I should be able to confirm this is not an issue with 1.14 by simply running a scraper and looking at the HTML in the ZIM? If I don't see the `display: none` it's fixed?

**Comment by audiodude** [comment]
> @Jaifroid So I should be able to confirm this is not an issue with 1.14 by simply running a scraper and looking at the HTML in the ZIM? If I don't see the `display: none` it's fixed?

@Jaifroid Sorry, I misread the linked issue, this is already confirmed to be happening in a ZIM created with 1.14 yes?

**Comment by Jaifroid** [comment]
Yes, it was with a Wikivoyage ZIM @kelson42 ran for me yesterday. Let me check the meta data...

**Comment by Jaifroid** [comment]
Well, the metadata of this 2024-08 English Wikivoyage say 1.13, but it must be using the new endpoint because all the stylesheets are different, and it's a very different ZIM from the last available version (2024-06)... And having the sections all force-hidden was not an issue with all previous Wikivoyage scrapes.

![image](https://github.com/user-attachments/assets/c60472c5-80b8-455c-9add-6dc6441e7f86)

**Comment by Jaifroid** [comment]
Issue also manifesting in the Chrome extension in ServiceWorkerLocal mode (as well as in Restricted mode). In SW-local mode, inline JS is blocked by the Chrome extension APIs, and so the following script in the HTML of each page is blocked:

![image](https://github.com/user-attachments/assets/6f7d61ed-dbb3-4230-94ae-1fb2eea6513e)

(Not 100% sure that's the issue, but it's the only inline script I've found.) This inline script must be calling a function defined in an attached script, and that function probably traversed the DOM for hidden sections and unhides them.

This behaviour will be in the code coming from the endpoint, I doubt it's been introduced by someone in mwOffliner, though it's possible (and has happened in the past). All that's needed is code in mwOffliner to remove the style display rule on these nodes before the HTML is stored in the ZIM, assuming we always want all sections to be unhidden by default, and assuming we don't want to replicate the old click action which opened and closed sections using details-summary tags.

**Comment by THEBOSS0369** [comment]
Hey @Jaifroid @audiodude ! I think i can work on this issue but before i come with up with any approach, i was trying to setup this repo locally and i'm getting this error while running `npm ci` i have switched to node `v18.20.4 `using `nvm use 18` . I was following contributing.md file 
this is the error i'm getting 
![Image](https://github.com/user-attachments/assets/9467f30e-a2f7-4277-9c67-c664a9fa644b)
![Image](https://github.com/user-attachments/assets/e39ba55b-51f5-45df-ad7a-f4f99297428c)

Thanks!

**Comment by audiodude** [comment]
@THEBOSS0369 see https://github.com/openzim/mwoffliner/issues/2029

**Comment by audiodude** [comment]
> I just checked the latest 2024-10 ZIM of Wikivoyage that [@audiodude](https://github.com/audiodude) ran for me the other day, and this issue seems no longer to be present. [@audiodude](https://github.com/audiodude) Can you confirm that you fixed it and this issue can be closed?

No I didn't fix it, I would have linked to this issue if I did. Possibly the output of mediawiki changed?

**Comment by Jaifroid** [comment]
> No I didn't fix it, I would have linked to this issue if I did. Possibly the output of mediawiki changed?

Yes, I was sure you would have! So maybe fixed at source. I think we should verify rather than go on one ZIM, if we have any ZIMs other than Wikivoyage currently produced from dev.

**Comment by THEBOSS0369** [comment]
@audiodude ! Can you look into this i have fixed the npm ci error and shared the steps for it here #2029 but while trying to start the server this error is coming i tried to solve it but ain't to able to resolve it .
![Image](https://github.com/user-attachments/assets/962bbec9-6b5f-4980-b66b-6d39a0329193)

If you have any solution for this it will be very helpful
Also do i have to add the email everytime or i can leave it?

**Comment by THEBOSS0369** [comment]
Hey @audiodude ! can you guide me how to fix this error i haven't been able to fix this yet i tried it from stack overflow and chat gpt but still ain't able to fix this. Your help will be appreciated. Thanks!

**Comment by audiodude** [comment]
Have you tried the command listed in the README? `npm run mwoffliner`

**Comment by audiodude** [comment]
@Jaifroid Are these the `section` tags that had `display: none`?

![Image](https://github.com/user-attachments/assets/a45114c3-ce26-4e71-824b-4d881b552f7d)

This is with a ZIM I created with 1.14 a while ago, `wikipedia_bm_all_maxi_2024-08.114.zim`.

![Image](https://github.com/user-attachments/assets/28c2a78b-2e36-46a6-bc8c-ceccbd423c94)

**Comment by Jaifroid** [comment]
@audiodude I'm guessing your snapshot is taken after the included JS is run to remove `style="dispay: none;`. It certainly looks like something has removed whatever was in the `style` attributes and has just left empty attributes. My point is about the raw HTML before any client-side manipulation of it has occurred.

Looking at the latest `wikipedia_bm_all_nopic_2024-10.zim` scraped with 1.14.0-dev, the issue is still present. E.g. in the "Mali" article, you can see all subsequent sections after section 0 are hidden in the raw HTML, and the article is visually missing all the detail thereafter:

![Image](https://github.com/user-attachments/assets/3d73d538-8795-4dc0-ace3-ac1ab2c0c51b)

You can test this in the Kiwix JS browser extension (available from the Chrome or Firefox Add-On stores). This will initially offer to open the ZIM in restricted mode for security reasons. Accept this, and then the issue can easily be seen in that small ZIM.

My point is that the logic needs to be reversed: the `style="none;"` declarations should be removed at scrape time, so they are not in the ZIM. The current functionality is useless in any case, because headings can't be clicked to open or close them even in ServiceWorker mode (or in any ZIM reader that runs client-side JS from the ZIM). If we want to implement that functionality, we need to add our own summary-details tags at scrape time, which was the previous solution, or implement that in CSS only. If we have decided to go with all headings always open even on mobile devices (a decision I wouldn't disagree with, especially when all readers have an easily accessible Table of Contents for ZIM articles), then I would suggest removing the `style="display: none;"` attributes at scrape time and also the redundant script that unhides sections client-side.

**Comment by Jaifroid** [comment]
I've opened [a separate issue](https://github.com/openzim/mwoffliner/issues/2096) for the problem mentioned in https://github.com/openzim/mwoffliner/issues/2074#issuecomment-2267617460. I confirm this exists in the latest `wikipedia_bm_all_nopic_2024-10.zim`.

**Comment by THEBOSS0369** [comment]
> Have you tried the command listed in the README? `npm run mwoffliner`

Yes sir I tried, it wasn't working I found that this is happening due to not starting of redis server which was happening because GitHub codespaces wasn't allowing me. I will try to set up repo locally but not sure I will be able to do that due to less space in my system 😓

**Comment by audiodude** [comment]
Fixed by #2097
